src/pages/index.js

```diff
@@ -8,62 +8,210 @@ import {
   const cardElement = new Card(
     {
       data,
       handleImageClick: (imageData) => {
         cardPreview.open(imageData);
       },
+      handleDeleteClick: () => {
+        deleteModal.setSubmitAction(() => {
+          deleteModal.renderLoading(true);
+          const id = cardElement.getId();
+          api
+            .deleteCard(id)
+            .then(() => {
+              cardElement.deleteButton();
+              deleteModal.close();
+            })
+            .catch((err) => {
+              console.error(err);
+            })
+            .finally(() => {
+              deleteModal.renderLoading(false);
+            });
+        });
+        deleteModal.open();
+      },
+
+      handleLikeClick: () => {
+        const id = cardElement.getId();
+        if (cardElement.isLiked()) {
+          api
+            .unLikeCard(id)
+            .then((data) => {
+              cardElement.setLikes(data.likes);
+            })
+            .catch((err) => console.error(err));
+        } else {
+          api
+            .likeCard(id)
+            .then((data) => {
+              cardElement.setLikes(data.likes);
+            })
+            .catch((err) => console.error(err));
+        }
+      },
+      myId: userId,
     },
     selectors.cardTemplate
   );
+
   const newCard = cardElement.getView();
   cardSection.addItem(newCard);
 };
 
-const cardSection = new Section(
-  {
-    items: initialCards,
-    renderer: renderCard,
-  },
-  selectors.cardSection
-);
-cardSection.renderItems();
+const deleteModal = new PopupWithConfirmation({
+  popupSelector: cardDeleteModal,
+  loadingText: "Deleting...",
+});
 
 // User Info
-const user = new UserInfo(".profile__title", ".profile__subtitle");
+const user = new UserInfo(
+  ".profile__title",
+  ".profile__subtitle",
+  ".profile__image"
+);
+
+//pulling in the card data via API
+let cardSection;
+let userId;
+
+Promise.all([api.getUserInfo(), api.getCardList()])
+  .then(([userData, data]) => {
+    user.setUserInfo({
+      title: userData.name,
+      subtitle: userData.about,
+    });
+    user.setAvatarInfo(userData.avatar);
+    userId = userData._id;
+    cardSection = new Section(
+      {
+        items: data,
+        renderer: renderCard,
+      },
+      selectors.cardSection
+    );
 
-// Popup with form
+    cardSection.renderItems();
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+
+// Popup with form New Card Creation
 const newCardPopup = new PopupWithForm({
   popupSelector: "#add-card-modal",
   handleFormSubmit: (inputValues) => {
-    renderCard(inputValues);
-    newCardPopup.close();
+    newCardPopup.renderLoading(true);
+    api
+      .addCard(inputValues)
+      .then((inputValues) => {
+        renderCard(inputValues);
+        newCardPopup.close();
+      })
+      .catch((err) => {
+        console.error(err);
+      })
+      .finally(() => {
+        // Hide the loading state
+        editProfileModal.renderLoading(false);
```

Comment:
You should change the button state in the card popup

Reply:
No, here you use the edit profile popup `editProfileModal`, but you should use the card popup

Reply:
done